pkg/executor/mem_reader.go

```diff
@@ -199,7 +212,10 @@ func (m *memIndexReader) decodeIndexKeyValue(key, value []byte, tps []*types.Fie
 		physTblIDColumnIdx = m.outputOffset[m.physTblIDIdx]
 	}
 
-	ds := make([]types.Datum, 0, len(m.outputOffset))
+	if m.resultRows == nil {
+		m.resultRows = make([]types.Datum, 0, len(m.outputOffset))
+	}
```

Comment:
Maybe we can init it when create `memIndexReader`

Reply:
done